nidmviewerfsl/callSlicer.py

```diff
@@ -0,0 +1,195 @@
+# the same size (more blank space is included at the edges/sides). In order to display a slice view this
+# extra blank space at the side of the nifti must be accounted for (else the FSL template and SPM
+# excursion set will not align in the slice display) and to do this, we must resize the SPM nifti by adding
+# more blank space to the side (note this does not affect the statistic values inside the excursion set).
+#
+# To resize an SPM map to match the layout of an FSL template the FSL function `flirt` can be used. However,
+# to do this specific transform a resize matrix is required.
+#
+# By default, if resizing a smaller map to a larger map `flirt` creates an empty map of the same size
+# as the larger map and places the smaller map in the front-left hand corner, centered in the z-axis.
+#
+# In other words when the smaller map is enlarged, (x, y, z) in the original small map becomes
+# (x, y, {l_z-s_z}/2 + z) in the larger map where l_z is the z dimension of the larger map and s_z
+# is the z dimension of the smaller map.
+#
+# This means the z dimensions of the SPM brain volume is now aligned with the z dimension of the FSL
+# template brain volume but the x and y dimensions are not. To rectify this the following transform
+# matrix must be used in flirt:
+#
+#  / 1 0 0 dx \
+# |  0 1 0 dy  |
+# |  0 0 1 0   |
+#  \ 0 0 0 1  /
+#
+# Where dx = l_x - s_x, the difference in size of the x dimensions.
+# and dy = l_y - s_y, the difference in size of the y dimensions.
+#
+#-------------------------------------------------------------------------------------------------------
+#
+# When the voxel size in mm in both maps is not the same:
+#
+# When the voxel size in mm is not the same a scaling factor must be added. When the SPM map has a
+# voxel size larger than 2, the matrix to scale and align the SPM map to the FSL 2mm template
+# simplifies to:
+#
+#  / 1 0 0 s*dx \
+# |  0 1 0 s*dy  |
+# |  0 0 1  0    |
+#  \ 0 0 0  1   /
+#
+# where s = 1/v_spm where v_spm is the voxel size of the SPM map.
+#
+#-------------------------------------------------------------------------------------------------------
+# Source: https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FLIRT/FAQ
+# Author: Tom Maullin (29/11/2017)
+
+import subprocess
+import os
+import shutil
+import random
+
+def nifDim(niftiFilename, k):
+    #Retrieve the k dimension of a nifti given it's filename using FSL.
+    
+    if k == 'x':
+        arg = 'dim1'
+    elif k == 'y':
+        arg = 'dim2'
+    elif k == 'z':
+        arg = 'dim3'
+    elif k == 'pix':
+        arg = 'pixdim1'
+    else:
+        error('Enter a valid dimension... x, y, z or pix')
+
+    #Make the command
+    getDimString = "fslhd " + niftiFilename + " | cat -v | grep ^" + arg
+    #Run the command
+    subprocess.check_call(getDimString, shell=True, stdout=subprocess.PIPE)
+    process = subprocess.Popen(getDimString, shell=True, stdout=subprocess.PIPE)
```

Comment:
Using `shell=True` can be a big security hazard, and only done if you are absolutely certain some uninitialized input is impossible. See https://docs.python.org/2/library/subprocess.html#frequently-used-arguments

Reply:
Thanks @vsoch! @TomMaullin: can you have a look into this? Thanks

Reply:
Hi @vsoch ,

Ah thanks! Yes I think I can resolve this for a lot of the script. The reason I am not sure how to work around this completely, however, is that this code needs access to the variable `${FSLDIR}` to work out where to access the FSL MNI 2mm template (and also in another branch this is also used to find the FSL CSS style sheet, logo image and background image). I am not aware of a way this variable can be retrieved without setting `shell=True` somewhere down the line. Do you know of a way I could work around this?

@satra I actually originally wrote this code with nibabel and nilearn! But we decided that it would be best to try and use the FSL commands as much as possible and moved away from this idea. I could rewrite this with nibabel but would have the same problem as if I were to write this command with `shell=False`. I believe the only issue is that to find the template nifti filename would require the `${FSLDIR}` variable which needs to be retrieved from the shell.